arch/sim/src/sim/up_releasepending.c

```diff
@@ -114,7 +118,11 @@ void up_release_pending(void)
 
           /* Then switch contexts */
 
+#ifdef CONFIG_SIM_PREEMPTIBLE
+          swapcontext(&prev_rtcb->xcp.context, &rtcb->xcp.context);
```

Comment:
Why not use sigsetjmp and siglongjmp:
https://linux.die.net/man/3/sigsetjmp

Reply:
By looking more at this API `sigsetjmp/siglongjmp` doesn't have a direct way for setting the initial task context. The ucontext API has the advantage of having the makecontext function. We need to manually change the PC and the SP in the `sigjmp_buf` from the `up_initial_state` call and this should be done for every arch.